aulas/modulo5/bloco14/projeto-labecommerce/src/index.ts

```diff
@@ -0,0 +1,173 @@
+import app from "./app";
+import express, { Response, Request } from "express";
+
+import connection from "./connection";
+import { CreateProduct, CreateUser, GetProductsByUserId, RegisterPurchase } from "./utils/functions";
+import { prod, purchase, user } from "./utils/types";
+
+// TEST CONNECTION
+app.get('/ping', (req: Request, res: Response) => {
+    try {
+        res.status(200).send("Pong")
+    }
+    catch (err) {
+        res.status(500).end("Ocorreu algo de errado.")
+    }
+})
+
+// GET ALL USERS
+app.get('/users', async (req: Request, res: Response) => {
+    let errorCode = 500
+    try {
+        const users = await connection('labecommerce_users')
+            .select()
+
+        res.status(200).send(users)
+    } catch (err: any) {
+        res.status(errorCode).end(err.message)
+    }
+})
+
+// GET ALL PRODUCTS
+app.get('/products', async (req: Request, res: Response) => {
+    let errorCode = 500
+    try {
+        const products = await connection('labecommerce_products')
+            .select()
+
+        res.status(200).send(products)
+    } catch (err: any) {
+        res.status(errorCode).end(err.message)
+    }
+})
+
+// GET PURCHASES BY USER ID
+app.get('/users/:user_id/purchases', async (req: Request, res: Response) => {
+    let errorCode = 500
+    try {
+        const userId = Number(req.params.user_id)
+
+        const purchases = await GetProductsByUserId(userId)
+
+        res.status(200).send(purchases)
+    } catch (err: any) {
+        res.status(errorCode).end(err.message)
+    }
+})
+
+// CREATE USER
+app.post('/users', async (req: Request, res: Response) => {
+    let errorCode = 500
+    try {
+        const { name, email, password } = req.body
+
+        if (!name || !email || !password) {
+            errorCode = 400
+            throw new Error("Parâmetro necessário não enviado!")
+        }
+        if (name === "" || email === "" || password === "") {
+            errorCode = 400
+            throw new Error("Parâmetros não podem estar vazios.")
+        }
+
+        const checkDuplicate = await connection('labecommerce_users')
```

Comment:
Aqui nessa função, você poderia desestruturar o array, colocando [checkDuplicate], dessa forma sua validação precisaria ser apenas 

if(checkDuplicate){
}

Poderia usar essa conexão para fazer uma função de getUserByEmail e utilizar ela para essas validações. Se a desestruturação não funcionar, basta colocar no return da função a variável da consulta que você fez com um [0] 

return checkDuplicate[0], e aí na validação seria repetir o if ali de cima. Se não for undefined você imprime a mensagem de erro dizendo que já existe um usuário com aquele email.